### PR TITLE
feature(Set Number of months for CMM)

### LINKED ISSUE #4151: Allow users to set the number of months used to calculate the CMM
Currently, the CMM is calculated using [six months of data](https://github.com/IMA-WorldHealth/bhima/blob/master/server/controllers/stock/core.js#L416), however, the algorithm allows us to specify a different base range of months of data.  Our friends in Vanga would like a way to input their own number of months of data for calculating the CMM, since they haven't been running the system for 6 months yet.

### PR DESCRIPTION
**Pull request by lomamech**
- Add in Enterprise module the possibility for the user to set number of
  months used to the calcul of CMM

- Update stock core and using the number from enterprise setting
- (Test) add property month_average_consumption in Integration test

closes #4151

### CODE REVIEW

**Comment by kwilu** [comment]
Hello @lomamech, it seems like you have referenced #4151 in your pull request description, but you have not referenced them in your commit message description(s). When you reference an issue in a commit message, it automatically closes the corresponding issue when the commit is merged.

Please run `git commit --amend` in your command line client to amend your commit message description with `Fixes #4151.`.

An example of a correctly-formatted commit:
```
commit fabd5e450374c8dde65ec35f02140383940fe146
Author: kwilu
Date:   Sat Mar 18 13:42:40 2017 -0700

    pull requests: Check PR commits reference when issue is referenced.

    Fixes #51.
```

Thank you for your contributions to BHIMA!

File: client/src/modules/enterprises/enterprises.html [hunk 1]
```diff
@@ -217,6 +217,18 @@
             <i class="fa fa-cogs"></i> <span translate>ENTERPRISE.SETTINGS.TITLE</span>
           </div>
           <div class="panel-body">
+              <div class="form-group" ng-class="{ 'has-error' : EnterpriseForm.$submitted && EnterpriseForm.month_average_consumption.$invalid }">
+                <label class="control-label" translate>
+                  FORM.LABELS.DEFINE_MONTH_AVERAGE_CONSUMPTION
+                </label>
+                <input type="number" class="form-control" autocomplete="off"
+                  name="month_average_consumption" required ng-model="EnterpriseCtrl.enterprise.settings.month_average_consumption"
```
Comment (jniles) [inline client/src/modules/enterprises/enterprises.html]: Does it need to be required?  Or do you set it to a default value in the enterprise controller?  There are some institutions that won't use stock management, so we shouldn't force them to set this variable unnecessarily.

File: server/models/migrations/next/migrate.sql [hunk 2]
```diff
@@ -1,3 +1,5 @@
 /*
 Migrate to next
  */
+-- @lomamech 2020-03-24
+ALTER TABLE `enterprise_setting` ADD COLUMN `month_average_consumption` SMALLINT(5) NOT NULL DEFAULT 6;
```
Comment (jniles) [inline server/models/migrations/next/migrate.sql:5]: 👍

File: test/integration/enterprises.js [hunk 3]
```diff
@@ -28,7 +28,7 @@ describe('(/enterprises) Enterprises API', () => {
     phone : '00904940950932016',
     location_id : '1f162a10-9f67-4788-9eff-c1fea42fcc9b',
     currency_id : 1,
-    settings : { enable_price_lock : 0 },
+    settings : { enable_price_lock : 0, month_average_consumption : 10 },
```
Comment (jniles) [inline test/integration/enterprises.js]: Can you check that the GET request returns this value?

Can you also write a test that shows it properly sets the default value?

**Review by jniles — COMMENTED** [review]
This looks good to me.  The only change I would like to see are the two integration tests:
  1. Check to see that month_average_consumption is being read from the database on GET.
  2. Make sure a default value is set when you create an enterprise.

Otherwise, looks excellent!  This will be a huge help.

**Comment by kwilu** [comment]
Hello @lomamech, it seems like you have referenced #4151 in your pull request description, but you have not referenced them in your commit message description(s). When you reference an issue in a commit message, it automatically closes the corresponding issue when the commit is merged.

Please run `git commit --amend` in your command line client to amend your commit message description with `Fixes #4151.`.

An example of a correctly-formatted commit:
```
commit fabd5e450374c8dde65ec35f02140383940fe146
Author: kwilu
Date:   Sat Mar 18 13:42:40 2017 -0700

    pull requests: Check PR commits reference when issue is referenced.

    Fixes #51.
```

Thank you for your contributions to BHIMA!

**Comment by jniles** [comment]
bors try

**Pull request merged**